lib/grandpa/grandpa.go

```diff
@@ -163,7 +163,7 @@ func NewService(cfg *Config) (*Service, error) {
 		justification:      make(map[uint64][]*SignedPrecommit),
 		head:               head,
 		in:                 make(chan GrandpaMessage, 128),
-		resumed:            make(chan struct{}),
+		resumed:            make(chan struct{}, 100),
```

Comment:
you can leave this as unbuffered since it's a channel that's just created and closed on resume

Reply:
Updated.